edx_event_bus_kafka/internal/tests/test_consumer.py

```diff
@@ -88,46 +102,58 @@ def setUp(self):
     @patch('edx_event_bus_kafka.internal.consumer.logger', autospec=True)
     def test_consume_loop_disabled(self, mock_logger):
         self.event_consumer.consume_indefinitely()  # returns at all
-        mock_logger.error.assert_called_once_with("Kafka consumers not enabled")
+        mock_logger.error.assert_called_once_with("Kafka consumers not enabled, exiting.")
 
     @override_settings(
         EVENT_BUS_KAFKA_SCHEMA_REGISTRY_URL='http://localhost:12345',
         EVENT_BUS_KAFKA_BOOTSTRAP_SERVERS='localhost:54321',
         EVENT_BUS_TOPIC_PREFIX='prod',
     )
-    def test_consume_loop(self):
+    @patch('edx_event_bus_kafka.internal.consumer.logger', autospec=True)
+    def test_consume_loop(self, mock_logger):
         """
         Check the basic loop lifecycle.
         """
-        poll_call_count = 0
-
-        def fake_poll(*args, **kwargs):
-            nonlocal poll_call_count
-            poll_call_count += 1
-            # Return normally twice (to show looping), then throw (to
-            # show that we're not actually handling exceptions, at
-            # least at the moment.) If we start suppressing
-            # exceptions, we'll need some other way to break the loop
-            # for this test.
-            if poll_call_count >= 3:
+        emit_call_count = 0
+
+        def fake_emit(*args, **kwargs):
+            nonlocal emit_call_count, self
+            emit_call_count += 1
+            # Perform the following sequence of events on successive calls:
+            # 1) accept a message normally, 2) throw, 3) accept a message normally
+            # again, and 4) stop the loop so we can finish the unit test.
```

Comment:
Nit: This might make more sense as a docstring?

Reply:
Yeah. Although... I've just discovered that `side_effect` can take a *list* of functions to call, which would allow me to get rid of the manual sequencing: https://docs.python.org/3.8/library/unittest.mock.html#unittest.mock.Mock.side_effect

Reply:
Oh... no luck. Emphasis mine:

> If side_effect is an iterable then each call to the mock will **return** the next value from the iterable.